lib/routes/espn/news.ts

```diff
@@ -0,0 +1,80 @@
+import { Route } from '@/types';
+import { load } from 'cheerio';
+import cache from '@/utils/cache';
+import ofetch from '@/utils/ofetch';
+import Parser from 'rss-parser';
+
+import dayjs from 'dayjs';
+import timezone from 'dayjs/plugin/timezone';
+import utc from 'dayjs/plugin/utc';
+import 'dayjs/locale/en';
+dayjs.locale('en');
+dayjs.extend(timezone);
+dayjs.extend(utc);
+
+const parser = new Parser();
+
+export const route: Route = {
+    path: '/news/:sport',
+    name: 'ESPN News Archive',
+    maintainers: ['GymRat102'],
+    example: '/news/nba',
+    categories: ['traditional-media'],
+    parameters: { sport: 'sport category, can be nba, nfl, mlb, nhl etc.' },
+    description: `Get the news feed of the sport you love on ESPN.
+| Sport                |  sport  |  Sport         |  sport  |
+|----------------------|---------|----------------|---------|
+| 🏀NBA                | nba     | 🎾Tennis       | tennis  |
+| 🏀WNBA               | wnba    | ⛳️Golf         | golf    |
+| 🏈NFL                | nfl     | 🏏Cricket      | cricket |
+| ⚾️MLB                | mlb     | 🏍️Motorsports  | rpm     |
+| 🏒NHL                | nhl     | 🏎️F1           | f1      |
+| ⛹️College Basketball | nhl     | 🥊MMA          | mma     |
+| 🏟️️College Football   | nhl     | 🏈UFL          | uffl    |
+| 📆NCAA               | ncaa    | 🏉Rugby        | rugby   |
+| 📆NCAA Woman         | ncaaw   | 🃏Poker        | poker   |
+| ⚽️Soccer             | soccer  |                |         |`,
+    radar: [
+        {
+            source: ['espn.com/:sport*'],
+            target: '/news/:sport',
+        },
+    ],
+    handler: async (ctx) => {
+        const { sport = 'nba' } = ctx.req.param();
+        const response = await ofetch(`https://www.espn.com/espn/rss/${sport}/news`);
+        const feed = await parser.parseString(response);
+
+        const list = feed.items.map((item) => ({
+            title: item.title ?? '',
+            link: item.link ?? '',
+            author: item.creator,
+            guid: item.guid,
+        }));
+
+        const items = await Promise.all(
+            list.map((item) =>
+                cache.tryGet(item.link, async () => {
+                    const response = await ofetch(item.link);
+                    const $ = load(response);
```

Comment:
Each article can be obtained using API as well if you provide search parameters

Reply:
Hi @TonyRL, I was able to fetch the feed using suggested API, but as fat as obtaining the content of each content, I could not identify a parameter of the same API that can be set to provide full text of an article.

This request from `https://www.espn.com/nba/` when scrolling down provides the following parameter:

```
https://onefeed.fan.api.espn.com/apis/v3/cached/contentEngine/oneFeed/leagues/nba
    ?source=ESPN.com%2B-%2BFAM
    &showfc=true
    &region=us
    &offset=55
    &limit=15
    &lang=en
    &authorizedNetworks=espn_free
    &editionKey=espn-en
    &device=desktop
    &pubkey=espn-en-nba-index
    &isPremium=true
    &locale=hk
    &featureFlags=expandAthlete
    &featureFlags=mmaGB
    &featureFlags=enhancedGameblock
    &featureFlags=watchRowLeagueAiring
    &featureFlags=overrideMetadata
    &showAirings=buy,live
```
but none of them are related to full text extraction. And this API seems an undocumented one, I am not able to find the parameter you mentioned.

But for extracting full context, I get a slightly better way: for each article link of feed I get, I can use `?partial=article` parameter to only get the content of `div.article-body` instead of the whole page. The improvement is just about bandwidth saving, and the method for getting content is still DOM selection by my judgement.

I'd like to know more about how to dig into that API for its parameter. Or if the second method I mentioned above is suitable. Thanks.

Reply:
The site returns an API response of an article when you visit the article link with search parameter `xhr=1`

Reply:
wow thanks, really curious about how to find this type of hidden stuff.

Based on my initial, the full content it provides in response is not valid HTML, e.g.

<img width="1222" alt="image" src="https://github.com/user-attachments/assets/22bbb9f8-e714-4703-b439-bc80a4f46782">

so still need some handling. I'll tweak in this direction.

Reply:
If you click on an article, the site will visit the article link with quite a lot of parameters
![image](https://github.com/user-attachments/assets/4030585c-1627-4330-bf97-69b0507e5170) and only `xhr` is useful in this case.

Reply:
really appreciate